result.go

```diff
@@ -183,6 +183,7 @@ type GroupKey interface {
 	Cols() []ColMeta
 	Values() []values.Value
 
+	Col(j int) ColMeta
```

Comment:
I would prefer we didn't modify this interface since it is already quite large. We might end up refactoring this in the future. Can you change the areas where this is used to be `key.Cols()[j]` instead?